ArcGISRuntimeSDKQt_CppSamples/Routing/RouteAroundBarriers/RouteAroundBarriers.h

```diff
@@ -0,0 +1,109 @@
+// [WriteFile Name=RouteAroundBarriers, Category=Routing]
+// [Legal]
+// Copyright 2020 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifndef ROUTEAROUNDBARRIERS_H
+#define ROUTEAROUNDBARRIERS_H
+
+#include "PolygonBarrier.h"
+#include "RouteParameters.h"
+#include "Stop.h"
+
+namespace Esri
+{
+namespace ArcGISRuntime
+{
+class GraphicsOverlay;
+class Map;
+class MapQuickView;
+class PictureMarkerSymbol;
+class RouteTask;
+class SimpleFillSymbol;
+}
+}
+
+#include <QObject>
+
+class RouteAroundBarriers : public QObject
+{
+  Q_OBJECT
+
+  Q_PROPERTY(Esri::ArcGISRuntime::MapQuickView* mapView READ mapView WRITE setMapView NOTIFY mapViewChanged)
+  Q_PROPERTY(bool addStops READ addStops WRITE setAddStops NOTIFY addStopsChanged)
+  Q_PROPERTY(bool addBarriers READ addBarriers WRITE setAddBarriers NOTIFY addBarriersChanged)
+  Q_PROPERTY(bool findBestSequence READ findBestSequence WRITE setFindBestSequence NOTIFY findBestSequenceChanged)
+  Q_PROPERTY(bool preserveFirstStop READ preserveFirstStop WRITE setPreserveFirstStop NOTIFY preserveFirstStopChanged)
```

Comment:
many of these Q_PROPERTYs can switch to use the MEMBER keyword instead of having explicit setters/getters that just update a member, they can instead just modify the member directly. We use this in other samples if you need examples. Details in this doc - https://doc.qt.io/qt-5/properties.html